GPU/Vulkan/FramebufferVulkan.cpp

```diff
@@ -52,45 +54,224 @@
+layout (location = 0) out vec4 fragColor;
+void main() {
+  fragColor = texture(sampler0, v_texcoord0);
+}
+)";
+
+static const char tex_vs[] = R"(#version 400
+#extension GL_ARB_separate_shader_objects : enable
+#extension GL_ARB_shading_language_420pack : enable
+layout (location = 0) in vec3 a_position;
+layout (location = 1) in vec2 a_texcoord0;
+layout (location = 0) out vec2 v_texcoord0;
+void main() {
+  v_texcoord0 = a_texcoord0;
+  gl_Position = vec4(a_position, 1.0);
+}
+)";
 
 void ConvertFromRGBA8888_Vulkan(u8 *dst, const u8 *src, u32 dstStride, u32 srcStride, u32 width, u32 height, GEBufferFormat format);
 
-void FramebufferManagerVulkan::ClearBuffer(bool keepState) {
-	// keepState is irrelevant.
-	if (!currentRenderVfb_) {
-		return;
+FramebufferManagerVulkan::FramebufferManagerVulkan(VulkanContext *vulkan) :
+	vulkan_(vulkan),
+	drawPixelsTex_(nullptr),
+	drawPixelsTexFormat_(GE_FORMAT_INVALID),
+	convBuf_(nullptr),
+	convBufSize_(0),
+	textureCache_(nullptr),
+	shaderManager_(nullptr),
+	resized_(false),
+	pixelBufObj_(nullptr),
+	currentPBO_(0),
+	curFrame_(0),
+	pipelineBasicTex_(VK_NULL_HANDLE),
+	pipelinePostShader_(VK_NULL_HANDLE),
+	vulkan2D_(vulkan) {
+
+	// Create a bunch of render pass objects, for normal rendering with a depth buffer,
+	// with and without pre-clearing of both depth/stencil and color, so 4 combos.
+	VkAttachmentDescription attachments[2] = {};
+	attachments[0].format = framebufFormat;
+	attachments[0].samples = VK_SAMPLE_COUNT_1_BIT;
+	attachments[0].loadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+	attachments[0].storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+	attachments[0].stencilLoadOp = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+	attachments[0].stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+	attachments[0].initialLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+	attachments[0].finalLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+	attachments[0].flags = 0;
+
+	attachments[1].format = vulkan_->GetDeviceInfo().preferredDepthStencilFormat;
+	attachments[1].samples = VK_SAMPLE_COUNT_1_BIT;
+	attachments[1].loadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+	attachments[1].storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+	attachments[1].stencilLoadOp = VK_ATTACHMENT_LOAD_OP_LOAD;
+	attachments[1].stencilStoreOp = VK_ATTACHMENT_STORE_OP_STORE;
+	attachments[1].initialLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+	attachments[1].finalLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+	attachments[1].flags = 0;
+
+	VkAttachmentReference color_reference = {};
+	color_reference.attachment = 0;
+	color_reference.layout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+
+	VkAttachmentReference depth_reference = {};
+	depth_reference.attachment = 1;
+	depth_reference.layout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+
+	VkSubpassDescription subpass = {};
+	subpass.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
+	subpass.flags = 0;
+	subpass.inputAttachmentCount = 0;
+	subpass.pInputAttachments = NULL;
```

Comment:
Hmm, so this is where we could do programmable blending (it seems like always using a barrier, I wonder if / hope this translates well on tiled renderers):
https://www.khronos.org/registry/vulkan/specs/1.0/xhtml/vkspec.html#renderpass-feedbackloop

Would we need to list render to texture here?

-[Unknown]